src/pypromice/postprocess/csv2bufr.py

```diff
@@ -398,18 +405,24 @@ def fetch_old_positions(df, stid, time_limit, positions):
     # to the resulting single array. Otherwise, we can have jumps when the GPS data goes out
     # or comes back. The message coordinates can sometimes all be 0.0, so we check for this.
     print('fetching old positions for {}'.format(stid))
-    df_limited = df.last(time_limit)
+    df_limited = df.last(time_limit).copy()
     print('last transmission: {}'.format(df_limited.index.max()))
 
     if ('msg_lat' in df_limited) and ('msg_lon' in df_limited):
         if (0.0 not in df_limited['msg_lat'].values):
             df_limited['gps_lat'] = df_limited['gps_lat'].combine_first(df_limited['msg_lat'])
         if (0.0 not in df_limited['msg_lon'].values):
             df_limited['gps_lon'] = df_limited['gps_lon'].combine_first(df_limited['msg_lon'])
-
-    df_limited = linear_fit(df_limited, 'gps_alt', 1, stid)
+    
     df_limited = linear_fit(df_limited, 'gps_lat', 6, stid)
     df_limited = linear_fit(df_limited, 'gps_lon', 6, stid)
+    df_limited = linear_fit(df_limited, 'gps_alt', 1, stid)
+    
+    # altitude is not transmitted in the iridium messages, so for some sites it
+    # is missing for as long back as 2012 (SCO_U). So if the last three months
+    # did not contain any data, we interpolate based on the entire data
+    if df_limited.gps_alt.isnull().all():
```

Comment:
> The main issue of providing elevation data to enable submitting to DMI is not solved. 

The main issue I am looking at is the missing elevations in AWS_locations.csv file. I am fine continuing to ignore stations with missing elevations from the BUFR processing. We need to ask DMI what they would prefer.

> If we pass the >2 days old condition [here](https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/c3a934f309e427e7d3fb667b01d6b0bf6122e6bd/bin/getBUFR#L197), then we are calling linear_fit but only passing the dataframe limited to previous 90 days. So if we are missing altitude for that entire period we have no way to extrapolate forward to get elevation data, and will then fail the min_data_check.

Exactly. If I'd like to update things so that elevation is being extrapolated before the min_data_check, then I'd add the following test:
https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/c4c9a069fbaea5339f8ecb824a1ee32ecc0b71ce/src/pypromice/postprocess/csv2bufr.py#L424-L425
directly in https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/fixing-missing-elevation/bin/getBUFR. But I am fine only fixing the AWS_positions.csv for now.

> I am wondering if can always do the extrapolation just using the last "chunk" of valid elevation data (last x months), rather than the full historical timeseries. We want to keep this as efficient as possible, and I am concerned about feeding the full historical data through the linear fit model.

For some stations, the GPS is out of order for several years. So for these sites the elevations will need to be extrapolated from the entire record. Since at the moment the entire historical data is being loaded, it doesn't cost extra to make that extrapolation. If we move to a more minimalistic processing loading only the last x months, then we'll need to find a way around it, e.g. saving the AWS' yearly positions (or average horizontal and vertical velocities) in a lighter file so that a linear extrapolation is always possible without loading all the historical data.